data_structures/stacks/postfix_to_prefix_conversion.py

```diff
@@ -0,0 +1,49 @@
+"""
+Functions:
+- postfix_to_prefix(expression): Converts a postfix expression to a prefix expression.
+"""
+
+from .stack import Stack
+
+
+def postfix_to_prefix(expression_str: str) -> str:
+    """
+    Converts a postfix expression to a prefix expression.
+
+    Args:
+        expression (str): A string representing a postfix expression.
+
+    Returns:
+        str: A string representing the equivalent prefix expression.
+
+    Raises:
+        IndexError: If the expression is invalid or contains an invalid operator.
```

Comment:
1. Could you add checks in your code for these cases rather than relying on `Stack`'s IndexError? That way we can have more useful error messages for this algorithm.
2. Could you add a doctest that tests an invalid expression and a doctest that tests an invalid operator?